app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -266,19 +266,24 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
 
     @Override
     protected void onFragmentFirstTimeVisible() {
+        _hlEditor.recomputeHighlighting(); // Run before setting scroll position
+        int line = -1;
+        int startPosition = _appSettings.getLastEditPosition(_document.path, 0);
+
         final Bundle args = getArguments();
-        int startPos = _appSettings.getLastEditPosition(_document.path, _hlEditor.length());
-        if (args != null && args.containsKey(Document.EXTRA_FILE_LINE_NUMBER)) {
-            final int lno = args.getInt(Document.EXTRA_FILE_LINE_NUMBER);
-            if (lno >= 0) {
-                startPos = TextViewUtils.getIndexFromLineOffset(_hlEditor.getText(), lno, 0);
-            } else {
-                startPos = _hlEditor.length();
+        if (args != null) {
+            if (args.containsKey(Document.EXTRA_FILE_LINE_NUMBER)) {
+                line = args.getInt(Document.EXTRA_FILE_LINE_NUMBER);
             }
         }
 
-        _hlEditor.recomputeHighlighting(); // Run before setting scroll position
-        TextViewUtils.setSelectionAndShow(_hlEditor, startPos);
+        if (line > -1) {
+            TextViewUtils.getIndexFromLineOffset(_hlEditor.getText(), line, 0);
+            TextViewUtils.setSelectionAndShow(_hlEditor, startPosition);
+        } else {
```

Comment:
line == -1 is 'last line' this is important for opening quicknote, todo etc etc.